api/server/httprest/server.go

```diff
@@ -0,0 +1,112 @@
+package httprest
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/runtime"
+)
+
+var _ runtime.Service = (*Server)(nil)
+
+// httpHandler is a functional interface that implements the http handler functionality.
+type httpHandler func(
```

Comment:
Need to see if I can just remove this outright or import directly into beacon node package for use